### PR TITLE
crushtool: set type 0 name "device" for --build option

### PR DESCRIPTION
**Pull request by Sandy4999**
The crushmap built by crushtool cannot be used directly as the name of type 0 is not specified.
We may decompile the crushmap(it will automatically add type 0) and then recompile it to make it usable, but that's extra effort.

Signed-off-by: Sangdi Xu <xu.sangdi@h3c.com>

### CODE REVIEW

**Comment by Sandy4999** [comment]
Simple test steps:
`crushtool -o crushmap --build --num_osds 3 root straw 0`
`crushtool -i crushmap --test --check 3`

**Comment by Sandy4999** [comment]
@liewegas Could you take a look at this minor crushtool fix, thank you!

File: src/tools/crushtool.cc [hunk 1]
```diff
@@ -617,6 +617,7 @@ int main(int argc, const char **argv)
       crush.set_item_name(i, "osd." + stringify(i));
     }
 
+    crush.set_type_name(0, "device");
```
Comment (liewegas) [inline src/tools/crushtool.cc]: How about "osd"? That's what the default OSDMap simple generator uses, and this function is also assuming "osd" naming above.

**Comment by Sandy4999** [comment]
@liewegas Sure, that's definitely a better name :D. I have also changed the default type 0 name in the decompile process, but I'm not sure if there needs any other cleanup work.

**Comment by liewegas** [comment]
Actually the second patch I'm not sure about.  Right now the "osd" naming is only used in the build method, so this keeps us consistent there, but leaves everything else with a generic "device".  For example the second patch breaks the cli unit tests that expect the old naming.  Let's just do the first patch?

**Comment by Sandy4999** [comment]
@liewegas Hmm, sorry I made it a bit confusing, but in fact it's the first patch that fails the unit test. Currently the behavior is that type 0 name is not specified when built, and then set as "device" in the decompile stage. However, if it's already named as "osd", decompile won't change it. So maybe the easiest way is just use "device" for build option or we have to modify the cli test

**Comment by Sandy4999** [comment]
@liewegas I will discard the second patch, but let's decide on the solution to --build opition first so that the bot may have an easier life. Any suggestion about the fix? Thank you!

**Comment by liewegas** [comment]
Sorry for the slow reply.  On second thought I think what you have here looks okay.  Thanks!

**Pull request merged**